spring-cloud-gateway-core/src/test/java/org/springframework/cloud/gateway/filter/factory/PreserveHostHeaderGatewayFilterFactoryTests.java

```diff
@@ -46,14 +47,17 @@
 @DirtiesContext
 public class PreserveHostHeaderGatewayFilterFactoryTests extends BaseWebClientTests {
 
+	@SuppressWarnings("unchecked")
 	@Test
 	public void preserveHostHeaderGatewayFilterFactoryWorks() {
-		testClient.get().uri("/headers").header("Host", "www.preservehostheader.org")
+		testClient.get().uri("/multivalueheaders").header("Host", "www.preservehostheader.org")
```

Comment:
Is there a reason why you changed the URI? This endpoint is behaving differently so I would not be sure if your change is not going to break someone else's application.
Can you verify if both endpoints are working in this test method?

Reply:
I also already created a PR to fix this: https://github.com/spring-cloud/spring-cloud-gateway/pull/1453

When you just use /headers, it only returns one value despite the fact that it actually has two or more values. Therefore, the former version of this test did not uncover the bug where /multivalueheaders would have.

Reply:
**/headers** and **/multivalueheaders** are endpoints defined within the unit tests of this project to echo back observed values for validation purposes.

The reason the duplicate bug was not caught by the existing tests is that **/headers** de-duplicates the headers anyway; Thus **/multivalueheaders** must be used to verify that there is exactly one Host header forwarded.

Reply:
Hah, yeah. What @dennis-menge said.

I can't believe I didn't see your PR, Dennis. I like your implementation better :)

Reply:
I think we already have three PRs for this issue right? :D

Reply:
yup, finally fixed some build problems, so this is next.